src/storage/mod.rs

```diff
@@ -686,6 +689,124 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
+                    .get(priority_tag)
+                    .inc();
+
+                let command_duration = tikv_util::time::Instant::now_coarse();
+
+                concurrency_manager.update_max_ts(max_ts);
+                // TODO: Use better error handling here
+                concurrency_manager
+                    .read_range_check(start_key.as_ref(), end_key.as_ref(), |key, lock| {
+                        Lock::check_ts_conflict(Cow::Borrowed(lock), &key, max_ts, &TsSet::Empty)
+                    })
+                    .map_err(mvcc::Error::from)?;
+
+                let snap_ctx = if need_check_locks_in_replica_read(&ctx) {
+                    let mut key_range = KeyRange::default();
+                    if let Some(start_key) = &start_key {
+                        key_range.set_start_key(start_key.as_encoded().to_vec());
+                    }
+                    if let Some(end_key) = &end_key {
+                        key_range.set_end_key(end_key.as_encoded().to_vec());
+                    }
+                    SnapContext {
+                        pb_ctx: &ctx,
+                        read_id: None,
+                        start_ts: max_ts,
+                        key_ranges: vec![key_range],
+                    }
+                } else {
+                    SnapContext {
+                        pb_ctx: &ctx,
+                        ..Default::default()
+                    }
+                };
+
+                let snapshot =
+                    Self::with_tls_engine(|engine| Self::snapshot(engine, snap_ctx)).await?;
+                {
+                    let begin_instant = Instant::now_coarse();
+                    let mut statistics = Statistics::default();
+                    let mut reader = MvccReader::new(
+                        snapshot,
+                        Some(ScanMode::Forward),
+                        !ctx.get_not_fill_cache(),
+                        ctx.get_isolation_level(),
+                    );
+                    let result = reader
+                        .scan_locks(
+                            start_key.as_ref(),
+                            end_key.as_ref(),
+                            |lock| lock.ts <= max_ts,
+                            limit,
+                        )
+                        .map_err(txn::Error::from);
+                    statistics.add(reader.get_statistics());
+                    let (kv_pairs, _) = result?;
+                    let mut locks = Vec::with_capacity(kv_pairs.len());
+                    for (key, lock) in kv_pairs {
+                        let mut lock_info = LockInfo::default();
+                        lock_info.set_primary_lock(lock.primary);
+                        lock_info.set_lock_version(lock.ts.into_inner());
+                        lock_info.set_key(key.into_raw().map_err(txn::Error::from)?);
+                        lock_info.set_lock_ttl(lock.ttl);
+                        lock_info.set_txn_size(lock.txn_size);
```

Comment:
Maybe better to use `into_lock_info`?

Reply:
Oh.. this is copied from the old code. I didn't even notice the problem before, there are many fields missing..